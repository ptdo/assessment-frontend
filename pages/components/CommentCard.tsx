import { Center, Box, Heading, Text, HStack } from '@chakra-ui/react';
import moment from 'moment'

import { IComment } from '../interfaces';

export default function CommentCard({comment} : {comment: IComment}) {
    return (
        <Center id={comment.id}>
            <Box
                border='solid'
                borderColor={'#356e6f'}
                boxShadow={'xl'}
                borderRadius={'20px'}
                p={6}
                boxSize={'95%'}
            >
            <HStack>
                <Heading fontSize={'xl'}>{comment.title}</Heading>
                <Text fontSize={'sm'}> Created:  {moment(comment.createdAt.toString()).format('MMMM Do, YYYY')}</Text>
            </HStack>
            <Text>{comment.description}</Text>
            </Box>
        </Center>
    )
}