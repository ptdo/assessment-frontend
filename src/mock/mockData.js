export const mockDate = new Date('2021-05-20T01:13:07.861Z')

export const mockPosts = [
  {
    title: "title 1",
    description: "Similique ipsum ut quae et cum. Quas et nisi at. Mollitia dolor quo dolores quia dolorem quam harum aut quis. Est enim vitae voluptate aliquid eligendi quia est doloremque.",
    createdAt: mockDate,
    id: "1",
    authors: [
      {
        name: "Alison Ondricka",
        avatar: "https://cdn.fakercloud.com/avatars/ricburton_128.jpg",
        id: "1",
      }
    ],
    comments: [
      {
        createdAt: mockDate,
        title: "Qui quo non omnis tenetur.",
        description: "Temporibus illo voluptatum illum possimus. Minus laudantium eum. Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.",
        id: "1",
      }
    ]
  },
  {
    title: "title 2",
    description: "Similique ipsum ut quae et cum. Quas et nisi at. Mollitia dolor quo dolores quia dolorem quam harum aut quis. Est enim vitae voluptate aliquid eligendi quia est doloremque.",
    createdAt: mockDate,
    id: "2",
    authors: [
      {
        name: "Alison Ondricka",
        avatar: "https://cdn.fakercloud.com/avatars/ricburton_128.jpg",
        id: "1",
      }
    ],
    comments: [
      {
        createdAt: mockDate,
        title: "Qui quo non omnis tenetur.",
        description: "Temporibus illo voluptatum illum possimus. Minus laudantium eum. Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.",
        id: "1",
      }
    ]
  },
  {
    title: "title 3",
    description: "Similique ipsum ut quae et cum. Quas et nisi at. Mollitia dolor quo dolores quia dolorem quam harum aut quis. Est enim vitae voluptate aliquid eligendi quia est doloremque.",
    createdAt: mockDate,
    id: "3",
    authors: [
      {
        name: "Alison Ondricka",
        avatar: "https://cdn.fakercloud.com/avatars/ricburton_128.jpg",
        id: "1",
      }
    ],
    comments: [
      {
        createdAt: mockDate,
        title: "Qui quo non omnis tenetur.",
        description: "Temporibus illo voluptatum illum possimus. Minus laudantium eum. Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.",
        id: "1",
      }
    ]
  },
  {
    title: "title 4",
    description: "Similique ipsum ut quae et cum. Quas et nisi at. Mollitia dolor quo dolores quia dolorem quam harum aut quis. Est enim vitae voluptate aliquid eligendi quia est doloremque.",
    createdAt: mockDate,
    id: "4",
    authors: [
      {
        name: "Alison Ondricka",
        avatar: "https://cdn.fakercloud.com/avatars/ricburton_128.jpg",
        id: "1",
      }
    ],
    comments: [
      {
        createdAt: mockDate,
        title: "Qui quo non omnis tenetur.",
        description: "Temporibus illo voluptatum illum possimus. Minus laudantium eum. Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.",
        id: "1",
      }
    ]
  },
  {
    title: "title 5",
    description: "Similique ipsum ut quae et cum. Quas et nisi at. Mollitia dolor quo dolores quia dolorem quam harum aut quis. Est enim vitae voluptate aliquid eligendi quia est doloremque.",
    createdAt: mockDate,
    id: "5",
    authors: [
      {
        name: "Alison Ondricka",
        avatar: "https://cdn.fakercloud.com/avatars/ricburton_128.jpg",
        id: "1",
      }
    ],
    comments: [
      {
        createdAt: mockDate,
        title: "Qui quo non omnis tenetur.",
        description: "Temporibus illo voluptatum illum possimus. Minus laudantium eum. Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.",
        id: "1",
      }
    ]
  },
  {
    title: "title 6",
    description: "Similique ipsum ut quae et cum. Quas et nisi at. Mollitia dolor quo dolores quia dolorem quam harum aut quis. Est enim vitae voluptate aliquid eligendi quia est doloremque.",
    createdAt: mockDate,
    id: "6",
    authors: [
      {
        name: "Alison Ondricka",
        avatar: "https://cdn.fakercloud.com/avatars/ricburton_128.jpg",
        id: "1",
      }
    ],
    comments: [
      {
        createdAt: mockDate,
        title: "Qui quo non omnis tenetur.",
        description: "Temporibus illo voluptatum illum possimus. Minus laudantium eum. Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.",
        id: "1",
      }
    ]
  }
]