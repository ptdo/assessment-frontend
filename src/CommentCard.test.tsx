import { render } from '@testing-library/react'

import CommentCard from '@pages/components/CommentCard'
import { mockPosts } from './mock/mockData'


describe('Test CommentCard Component', () => {
    const { container } = render(<CommentCard comment={mockPosts[0].comments[0]}/>)

    it('should render successfully', () => {
        expect(container.firstChild).not.toBeNull
    })

    it('should render expected format', () => {
        expect(container.firstChild).toMatchSnapshot(`
        <div>
            <div>
                <div>
                    <h2>Itaque quam molestias dignissimos minus nulla voluptatem ea fugiat quisquam.</h2>
                    <p> Created: May 19th, 2021</p>
                </div>
                <p>Temporibus illo voluptatum illum possimus. Minus laudantium eum. 
                Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.</p>
            </div>
        </div>
        `)
    })
})

