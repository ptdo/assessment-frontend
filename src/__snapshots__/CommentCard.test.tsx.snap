// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CommentCard Component should render expected format: 
        <div>
            <div>
                <div>
                    <h2>Itaque quam molestias dignissimos minus nulla voluptatem ea fugiat quisquam.</h2>
                    <p> Created: May 19th, 2021</p>
                </div>
                <p>Temporibus illo voluptatum illum possimus. Minus laudantium eum. 
                Corrupti provident blanditiis qui sed aperiam ut totam. Cupiditate autem dolor pariatur commodi. Nesciunt incidunt vero repellat reiciendis.</p>
            </div>
        </div>
         1`] = `null`;
